--- configure.in	17 Apr 2008 15:19:12 -0000	1.43
+++ configure.in	7 Sep 2009 08:50:17 -0000
@@ -468,7 +468,7 @@
 AC_CHECK_LIB(blkid, blkid_get_devname, [blkid_lib=yes], [blkid_lib=no], [-luuid])
 if test "$blkid_h" = yes -a "$blkid_lib" = yes; then
 	AC_DEFINE([HAVE_BLKID],1,[Define this if you have the blkid library.])
-	BLKID="-lblkid -luuid -ldevmapper -lselinux -lsepol"
+	BLKID="-lblkid -luuid"
 fi
 AC_SUBST(BLKID)
 

 	  	 
