diff -up dump-0.4b46/configure.ac.4~ dump-0.4b46/configure.ac
--- dump-0.4b46/configure.ac.4~	2020-10-01 22:05:29.253515108 +0200
+++ dump-0.4b46/configure.ac	2020-10-01 22:06:04.616692229 +0200
@@ -365,6 +365,22 @@ if test "$enable_zlib" != "no"; then
 	fi
 fi
 
+dnl Check for lzo headers and libraries
+dnl
+AC_CHECK_HEADER(lzo/minilzo.h, [minilzo_h=yes], [minilzo_h=no], [-])
+AC_CHECK_LIB(minilzo, lzo1x_1_compress, [minilzo_lib=yes], [minilzo_lib=no])
+if test "$minilzo_h" = yes -a "$minilzo_lib" = yes; then
+	if test "$STATICZ" = yes; then
+		LZOLIB="-Wl,-Bstatic -lminilzo -Wl,-Bdynamic"
+	else
+		LZOLIB="-lminilzo"
+	fi
+	AC_DEFINE([HAVE_LZO],1,[Define this if you have lzo library.])
+else
+	LZOLIB=""
+fi
+AC_SUBST(LZOLIB)
+
 dnl
 dnl Check for bzlib headers and libraries
 dnl
